File: lib_common-utils/src/main/java/com/common/utils/ByteShortUtils.java
package com.common.utils;

import java.nio.ByteBuffer;
import java.nio.ByteOrder;

public class ByteShortUtils {

    private ByteShortUtils() {
    }

    private static final class HolderClass {
        private static final ByteShortUtils instances = new ByteShortUtils();
    }

    public static ByteShortUtils getInstance() {
        return HolderClass.instances;
    }

    public short[] bytesToShort(byte[] bytes) {
        if (bytes == null) {
            return null;
        }
        short[] shorts = new short[bytes.length / 2];
        ByteBuffer.wrap(bytes).order(ByteOrder.LITTLE_ENDIAN).asShortBuffer().get(shorts);
        return shorts;
    }

    public byte[] shortToBytes(short[] shorts) {
        if (shorts == null) {
            return null;
        }
        byte[] bytes = new byte[shorts.length * 2];
        ByteBuffer.wrap(bytes).order(ByteOrder.LITTLE_ENDIAN).asShortBuffer().put(shorts);

        return bytes;
    }

}

File: lib_common-utils/src/main/java/com/common/utils/ByteShortUtils.kt
package com.common.utils

import java.nio.ByteBuffer
import java.nio.ByteOrder

object ByteShortUtils {

    @JvmStatic
    fun getInstance(): ByteShortUtils = this

    fun bytesToShort(bytes: ByteArray?): ShortArray? {
        if (bytes == null) {
            return null
        }
        val shorts = ShortArray(bytes.size / 2)
        ByteBuffer.wrap(bytes).order(ByteOrder.LITTLE_ENDIAN).asShortBuffer().get(shorts)
        return shorts
    }

    fun shortToBytes(shorts: ShortArray?): ByteArray? {
        if (shorts == null) {
            return null
        }
        val bytes = ByteArray(shorts.size * 2)
        ByteBuffer.wrap(bytes).order(ByteOrder.LITTLE_ENDIAN).asShortBuffer().put(shorts)
        return bytes
    }
}